### PR TITLE
Feat/branches table

### PR DESCRIPTION
**Pull request by dejan-crocoder**

### CODE REVIEW

File: migrations/tenant-db/0012_familiar_slyde.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ALTER TABLE transform_merge_requests ADD `target_branch` integer NOT NULL REFERENCES transform_branches(id);--> statement-breakpoint
+ALTER TABLE transform_merge_requests ADD `source_branch` integer NOT NULL REFERENCES transform_branches(id);--> statement-breakpoint
+CREATE UNIQUE INDEX `branches_name_idx` ON `transform_branches` (`name`);--> statement-breakpoint
+/*
+ SQLite does not support "Creating foreign key on existing column" out of the box, we do not generate automatic migration for that, so it has to be done manually
+ Please refer to: https://www.techonthenet.com/sqlite/tables/alter_table.php
+                  https://www.sqlite.org/lang_altertable.html
+
+ Due to that we don't generate migration automatically and it has to be done manually
+*/
```
Comment (dejan-crocoder) [inline migrations/tenant-db/0012_familiar_slyde.sql:17]: Issue: cant add not null column without default values, How do we resolve this ?

File: packages/functions/transform/src/merge-request-metrics.ts [hunk 2]
```diff
@@ -955,13 +971,25 @@ export async function run(extractMergeRequestId: number, ctx: RunContext) {
...
+  const { id: transformSourceBranchId } = await upsertBranch(ctx.transformDatabase, {
+    name: extractData.mergeRequest.sourceBranch || "",
+  })
+    .get();
+
+  const { id: transformTargetBranchId } = await upsertBranch(ctx.transformDatabase, {
+    name: extractData.mergeRequest.targetBranch || "",
+  })
+    .get();
+
```
Comment (dejan-crocoder) [inline packages/functions/transform/src/merge-request-metrics.ts:983]: Issue: Branches can not have names, maybe for deleted branches or something ? How do we treat null branch

File: migrations/tenant-db/0012_familiar_slyde.sql (hunk 1 above)
Comment (dejan-crocoder) [reply migrations/tenant-db/0012_familiar_slyde.sql:17]: pls check defintion of branches schema for target and source branches

Comment (dejan-crocoder) [reply migrations/tenant-db/0012_familiar_slyde.sql:17]: After further checking, I can see that source/target branches are not defined as nullable by their respective forge client libraries (octokit GitHub, gitbeaker -> Gitlab). Not sure what to do here

**Review by ivke995 — APPROVED** [review]
:1st_place_medal:

**Pull request merged**